services/galley/src/Galley/API/Internal.hs

```diff
@@ -243,10 +272,22 @@ internalAPI =
     mkNamedAPI @"status" (pure ())
       <@> mkNamedAPI @"delete-user" rmUser
       <@> mkNamedAPI @"connect" Create.createConnectConversation
+      <@> iTeamsAPI
       <@> mkNamedAPI @"upsert-one2one" iUpsertOne2OneConversation
       <@> mkNamedAPI @"feature-config-snd-factor-password-challenge" getSndFactorPasswordChallengeNoAuth
       <@> featureAPI
 
+iTeamsAPI :: API ITeamsAPI GalleyEffects
+iTeamsAPI = mkAPI $ \tid -> hoistAPIHandler id (base tid)
+  where
+    base :: TeamId -> API ITeamsAPIBase ITeamEffects
+    base tid =
+      mkNamedAPI @"get-team" (Teams.getTeamInternalH tid)
+        <@> mkNamedAPI @"create-binding-team" (Teams.createBindingTeamH tid)
+        <@> mkNamedAPI @"delete-binding-team-with-one-member" (Teams.internalDeleteBindingTeamWithOneMemberH tid)
+        <@> hoistAPI @IGetTeamNameAPI id (mkNamedAPI @"get-team-name" $ Teams.getTeamNameInternalH tid)
+        <@> hoistAPI @IUpdateTeamStatusAPI id (mkNamedAPI @"update-team-status" $ Teams.updateTeamStatusH tid)
```

Comment:
Using `hoistAPI` is probably not necessary here if the path routes (e.g. `"name"`) are nested inside `Named` combinator.

Reply:
True, and I have a commit that swizzles that around in my local development environment that will be included in the next round.